### ISSUE TITLE
A flat if/elif chain is reported as deep nesting

### ISSUE THREAD

**Issue by CognitiveDecisions** [rsionnach/sloppylint#25]
Python represents `elif` as a nested `If` inside `orelse`, and `visit_If` increments the depth counter for each one:

https://github.com/rsionnach/sloppylint/blob/e0e3ced/src/sloppy/analyzers/ast_analyzer.py#L83

So a flat dispatch with five branches reads as five levels of nesting.

### Repro

```python
def route(x):
    if x == 1:
        return 1
    elif x == 2:
        return 2
    elif x == 3:
        return 3
    elif x == 4:
        return 4
    elif x == 5:
        return 5
```

```
$ sloppylint n.py --format compact
  n.py:10  deep_nesting: Code nested 5 levels deep - consider refactoring
```

Nothing here is nested. The reported line (10) is the last `elif`, not a block anyone would indent.

### Effect

This lands hardest on exactly the code the rule is least useful for: flat state machines and dispatch tables, where an `elif` chain is the clearest form available.

### Suggested fix

In `visit_If`, walk the `orelse` chain without incrementing when the `orelse` is a single `If` that starts on a later line than its parent's body:

```python
def visit_If(self, node):
    orelse = node.orelse
    while len(orelse) == 1 and isinstance(orelse[0], ast.If):
        self._check_patterns(orelse[0])
        orelse = orelse[0].orelse
```

Then only genuine `else:` blocks containing an `if` add a level.

Related: #24 cannot be worked around today, because `--disable deep_nesting` does nothing.

Found on 0.5.1 (`e0e3ced`).
